packages/amplify-app/src/xcodeHelpers.js

```diff
@@ -15,42 +40,152 @@ async function getXcodeProjectDir() {
   return projDir;
 }
 
-async function addFileToXcodeProj(file, isResource) {
-  const projectPath = await getXcodeProjectDir();
-  // Silently return if not in same directory as xcode project
-  if (!projectPath) {
-    return;
+/**
+ * @private
+ * @param {object} project
+ * @param {string} name
+ * @returns {PBXGroupRef|undefined}
+ */
+function getGroupByName(project, name) {
+  /** @type PBXGroupRef */
+  let rootGroup = null;
+
+  const groups = Object.entries(project.hash.project.objects.PBXGroup);
+  for (let i = 0; i < groups.length; i++) {
+    const [key, value] = groups[i];
+    if (typeof value !== 'string') {
+      // only the root pbx group can have no name, path or description
+      const isRoot = name === undefined && value.name === undefined && value.path === undefined;
+      const matchesName = typeof name === 'string' && (value.name === name || value.path === name);
+
+      if (isRoot || matchesName) {
```

Comment:
Couldn't this lead to this function returning the root group when you're looking for a group that matches a certain name?

Reply:
No because `isRoot` checks if `name` is `unfefined` and `matchesName` checks if it's an actual `string`